src/components/ItemSelector/ItemSelector.js

```diff
@@ -14,21 +16,17 @@ const ItemSelector = () => {
     const [filter, setFilter] = useState('')
     const [items, setItems] = useState(null)
     const [maxOptions, setMaxOptions] = useState(new Set())
-    const { d2 } = useD2({})
+    const dataEngine = useDataEngine()
+    const debouncedFilterText = useDebounce(filter, 200)
```

Comment:
I think we should set 500 to get more out of it @jenniferarnesen @cooper-joe

Reply:
I would argue that 500ms is too long. For form validation the debounce can be longer, but for search the user is expecting to see results, so any delay is purely technical in nature. I would say 350ms should be the maximum, unless there's a strong technical backing to delay it further. (Related: [Doherty Threshold](https://lawsofux.com/doherty-threshold/))

Reply:
Good point, @cooper-joe. Let's go with 350 ms.